CMakeLists.txt

```diff
@@ -84,6 +84,7 @@ include(var_functions)
 option(NGRAPH_UNIT_TEST_ENABLE "Control the building of unit tests" TRUE)
 option(NGRAPH_TOOLS_ENABLE "Control the building of tool" TRUE)
 option(NGRAPH_CPU_ENABLE "Control the building of the CPU backend" TRUE)
+option(NGRAPH_INTELGPU_ENABLE "Control the building of the Intel GPU backend with clDNN" FALSE)
```

Comment:
`NGRAPH_INTELGPU`: Since this is based on CL, I suppose it will also support devices that can run CL, not just intel GPU. `NGRAPH_GPU` is not all GPU but only CUDA. maybe we should discuss about the name.

Reply:
I agree that the name should better be something like  NGRAPH_CL : for some RNN and some models of CPU, I have seen Intel OpenCL CPU drivers (for exple opencl-1.2-6.4.0.25) faster than Eigen : in other words that CL backend, could also be useful for some non gpu hardware.

Reply:
Propose exact name for this variable please.

Reply:
I suggest that we name the backends for the APIs that they are programmed to, not what is behind that API. Currently we have GPU which is really CUDA with an Nvidia GPU behind it and should really be called CUDA: NGRAPH_CUDA_ENABLE, and backend::create("CUDA")
This backend is for OpenCL, specifically targeting the Intel GPU but can be so much more. We could call it CL or OCL: NGRAPH_CL_ENABLE and backend::create("CL") or OCL or ???
Do we get support for OpenCL CPU drivers?
Do we get support for AMD GPUs?
This is not a simple decision and I don't want to make any changes without consideration.
I also don't want this PR stuck in limbo over a name. We can change it later and we already need to rename GPU.